argilla-sdk/src/argilla_sdk/datasets/_resource.py

```diff
@@ -73,15 +73,13 @@ def __init__(
             self._log_message(f"Settings dataset name to unique UUID: {name}")
 
         self.workspace_id = (
-            _model.workspace_id
-            if _model and _model.workspace_id
-            else self.__workspace_id_from_name(workspace=workspace)
+            _model.workspace_id if _model and _model.workspace_id else self._workspace_id_from_name(workspace=workspace)
         )
         self._model = _model or DatasetModel(
             name=name,
             workspace_id=UUIDUtilities.convert_optional_uuid(uuid=self.workspace_id),
         )
-        self._settings = self.__configure_settings_for_dataset(settings=settings)
+        self._settings = settings or Settings(_dataset=self)
```

Comment:
Here we need to assign the dataset to the incoming settings
```suggestion
        self._settings = settings or Settings(_dataset=self)
        self._settings.dataset = self
```

Reply:
Isn't this achieved by giving `Settings` the `_dataset` param?

Reply:
Sorry. I mean when the settings argument is provided.

Reply:
@frascuchon this might need to be handled still?